src/WindowsTracing/KrabsTracer.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "KrabsTracer.h"
+
+#include <optional>
+
+#include "ClockUtils.h"
+#include "EventTypes.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/ThreadUtils.h"
+
+namespace orbit_windows_tracing {
+
+using orbit_grpc_protos::SchedulingSlice;
+
+KrabsTracer::KrabsTracer(orbit_grpc_protos::CaptureOptions capture_options,
```

Comment:
I'd love to see a test for this class.

Reply:
I looked into it a little, it's not easily unit-testable. We'd have to essentially generate fake ETW data, which is a bit of work. We can definitely go for an integration test however, I have one locally. I'll dig deeper for the unit test, but this will be in another PR.

Reply:
Okay, thanks for looking into this.